lib/zclient.c

```diff
@@ -1461,6 +1462,223 @@ zebra_interface_vrf_update_read (struct stream *s, vrf_id_t vrf_id,
   *new_vrf_id = new_id;
   return ifp;
 }
+/**
+ * Connect to label manager in a syncronous way
+ *
+ * It first writes the request to zcient output buffer and then
+ * immediately reads the answer from the input buffer.
+ *
+ * @param zclient Zclient used to connect to label manager (zebra)
+ * @result Result of response
+ */
+int
+lm_label_manager_connect (struct zclient *zclient)
```

Comment:
You're right, this message is necessary. I'm just not sure if we need to send a response to the clients. We could do something similar to zread_hello() and ignore requests from invalid clients (proto >= ZEBRA_ROUTE_MAX || proto <= ZEBRA_ROUTE_STATIC) as well as any subsequent message whose client's protocol is not set (zero).